pkg/component/delete/undeploy_handler.go

```diff
@@ -40,14 +39,7 @@ func (o *undeployHandler) ApplyKubernetes(kubernetes v1alpha2.Component) error {
 	}
 	klog.V(4).Infof("Un-deploying the Kubernetes %s: %s", u.GetKind(), u.GetName())
 	// Un-deploy the K8s manifest
-	err = o.kubeClient.DeleteDynamicResource(u.GetName(), gvr.Resource.Group, gvr.Resource.Version, gvr.Resource.Resource)
-	if err != nil {
-		if strings.Contains(err.Error(), "not found") {
-			klog.V(3).Infof("Resource for %s not found; cause: %w", u.GetName(), err)
-			return nil
-		}
-	}
-	return err
+	return o.kubeClient.DeleteDynamicResource(u.GetName(), gvr.Resource.Group, gvr.Resource.Version, gvr.Resource.Resource)
```

Comment:
This is dangerous to change this behaviour. Because of `ownerReferences` and the garbage collection triggered by the deletion of a previous resource, this resource could have been deleted in the meantime, and we don't want to fail the command at this point in this case. Or the user could have deleted this resource by hand, and wants to delete the other ones with `odo delete`, etc

Reply:
Are you suggesting that in the case that the resource is not found for whatsoever reason, and this method returns a *NotFoundError, we return a `nil`?

Reply:
Yes, because we don't want to fail the complete command if the deletion of a single resource fails